client/src/components/SignUp.js

```diff
@@ -1,41 +1,96 @@
+        } = this.state
+
+        const validPassword = /^(?=.*[0-9])(?=.*[a-zA-Z])(?=\S+$).{6,}$/
+
+        const validEmail = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+
+        if(fName.localeCompare("") === 0 || lName.localeCompare("") === 0 ||
+            email.localeCompare("") === 0 || password.localeCompare("") === 0 ||
+            email.localeCompare(confirmEmail) !== 0 || password.localeCompare(confirmPassword)
+            || !validPassword.test(password)
+            || !validEmail.test(email)
+        ){
+            this.setState({renderAlert: true})
+        } else{
+            this.setState({renderAlert: false})
+            //creat new account!
+        }
+    }
+
     render() {
         return (
             <form>
 
+                {this.state.renderAlert && <Alert color="danger">
+                    Error! All fields must be filled in correctly
+                </Alert>
+                }
+
+
                 <h3>Sign Up</h3>
 
                 <div className="form-group">
                     <label>First Name</label>
-                    <input type="fname" className="form-control" id = "fname" placeholder="First name" />
+                    <input type="fname" name = "fName" onChange={this.handleFieldChange} className="form-control" id = "fname" placeholder="First name" />
                 </div>
                 <div className="form-group">
                     <label>Last Name</label>
-                    <input type="lname" className="form-control" id = "lname" placeholder="Last name" />
+                    <input type="lname" name = "lName" onChange={this.handleFieldChange} className="form-control" id = "lname" placeholder="Last name" />
                 </div>
 
                 <div className="form-group">
                     <label>Email</label>
-                    <input type="email" className="form-control" id = "email" placeholder="Enter email" />
+                    <input type="email" name = "email" onChange={this.handleFieldChange} className="form-control" id = "email" placeholder="Enter email" />
                 </div>
                 <div className="form-group">
                     <label>Confirm Email</label>
-                    <input type="email" className="form-control" id = "email" placeholder="Confirm email" />
+                    <input type="email" name = "confirmEmail" onChange={this.handleFieldChange} className="form-control" id = "email" placeholder="Confirm email" />
                 </div>
 
                 <div className="form-group">
                     <label>Password</label>
-                    <input type="password" className="form-control" id = "password" placeholder="Enter password" />
+                    <input type="password" name = "password" onChange={this.handleFieldChange} className="form-control" id = "password" placeholder="Enter password (Minimum six characters, at least one letter and one number)" />
```

Comment:
Something you might want to think about adding is the 'pattern' attribute. I used it for a different class before and it will pop up with a little bubble next to the field if the user doesn't enter the correct information. You can tell it how many characters you want it to be. Not sure about the letter and number requirement, but at least this way, the user has more of an idea of where they went wrong. Kind of like the 'required' attribute, which also might be a good idea to add.